src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -104,6 +67,8 @@ export async function makeCsharpConfig(config: SamLaunchRequestArgs): Promise<Sa
         globals: template?.Globals,
         properties: resource?.Properties,
     })
+    // TODO: walk the tree to find .sln, .csproj ?
+    config.codeRoot = getSamProjectDirPathForFile(config.samTemplatePath)
```

Comment:
this is equivalent to the `dirname()` call in 164473c34395 but now the decision lives in the CONFIG phase.